### PR TITLE
[Merged by Bors] - feat: Add lemmas about `Polynomial.scaleRoots`.

### PR DESCRIPTION
**Pull request by erdOne**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/Polynomial/ScaleRoots.lean [hunk 1]
```diff
@@ -158,6 +182,72 @@ theorem scaleRoots_aeval_eq_zero_of_aeval_div_eq_zero [Algebra R K]
...
+      | inl h => simp only [coeff_eq_zero_of_natDegree_lt h, zero_mul, mul_zero]
+      | inr hb =>
+        simp only [← e, mul_assoc, mul_comm (r ^ (_ - a)), ← pow_add]
+        rw [add_comm (_ - _), tsub_add_tsub_comm ha hb]
+
+lemma add_scaleRoots_of_natDegree_eq (p q : R[X]) (r : R)
+    (h : natDegree p = natDegree q) :
+    r ^ (natDegree p - natDegree (p + q)) • (p + q).scaleRoots r =
+      p.scaleRoots r + q.scaleRoots r := by
+  ext n; simp only [ge_iff_le, coeff_smul, coeff_scaleRoots, coeff_add, smul_eq_mul,
```
Comment (ericrbg) [inline Mathlib/RingTheory/Polynomial/ScaleRoots.lean:218]:
```suggestion
  ext n; simp only [coeff_smul, coeff_scaleRoots, coeff_add, smul_eq_mul,
```

some of these `simp?` suggestions are a bit broken right now sadly

File: Mathlib/RingTheory/Polynomial/ScaleRoots.lean [hunk 2]
```diff
@@ -158,6 +182,72 @@ theorem scaleRoots_aeval_eq_zero_of_aeval_div_eq_zero [Algebra R K]
   scaleRoots_eval₂_eq_zero_of_eval₂_div_eq_zero inj hr hs
 #align polynomial.scale_roots_aeval_eq_zero_of_aeval_div_eq_zero Polynomial.scaleRoots_aeval_eq_zero_of_aeval_div_eq_zero
 
+@[simp]
+lemma scaleRoots_mul (p : R[X]) (r s) :
+    p.scaleRoots (r * s) = (p.scaleRoots r).scaleRoots s := by
+  ext; simp [mul_pow, mul_assoc]
+
+lemma mul_scaleRoots (p q : R[X]) (r : R) :
+    r ^ (natDegree p + natDegree q - natDegree (p * q)) • (p * q).scaleRoots r =
```
Comment (ericrbg) [inline Mathlib/RingTheory/Polynomial/ScaleRoots.lean:191]: this deserves a docstring, and I wouldn't be against a `NoZeroDivisors` version of all the below. it took me a while to realise why we need the power of `r`, and I'm honestly not sure where stuff pops up (I think I just need to pick my favourite ring with zero divisors and run some examples to figure that out, though)

**Review by ericrbg — COMMENTED** [review]
looking pretty good!

**Review by Vierkantor — APPROVED** [review]
LGTM, thanks!

bors d+

**Comment by erdOne** [comment]
bors merge

**Pull request closed without merging**